python/cudf/cudf/core/column_accessor.py

```diff
@@ -371,6 +373,18 @@ def get_labels_by_index(self, index: Any) -> tuple:
             return self.names[start:stop:step]
         elif pd.api.types.is_integer(index):
             return (self.names[index],)
+        elif (bn := len(index)) > 0 and all(map(is_bool, index)):
+            if bn != (n := len(self.names)):
+                raise IndexError(
+                    f"Boolean mask has wrong length: {bn} not {n}"
+                )
+            if isinstance(index, (pd.Series, cudf.Series)):
```

Comment:
Would it be any better to replace this `if` with a `try`?

```python
try:
    index = list(index)  # assuming we want something we can coerce to a list
except <something>:
    raise NotImplementedError(...)  # or TypeError perhaps?
 ```

Reply:
In fact, one also can't use a Series as a boolean mask for `iloc` indexing of rows, but changing that would break too much for this PR, I think.

Reply:
Gotcha